### PR TITLE
Migrated Job app to CBV

### PR DESCRIPTION
**Pull request by amruthasangeeth**
Migrated function based views in Job application to class based views. The following are the views used:

ListView - To list the jobs
UpdateView - To edit the Job details
DeleteView - To delete a job object
DetailView - Displays the details of a job

### CODE REVIEW

**Comment by tapasweni-pathak** [comment]
cc @nida @smarshy.

Great if you both can have a look it it.

**Comment by smarshy** [comment]
@amruthasangeeth Based on build, the following checks/functionality seem to be missed:
1. On editing a job, there is a two way check that it does not fall out of the event range and also that its shifts don't fall out of the new job range
2. If you delete a job, there should be a check to see if there are associated shifts
3. There are simple error messages displayed under field values and those values should be retained as before

File: vms/job/views.py [hunk 1]
```diff
@@ -3,172 +3,115 @@
...
 
-@login_required
-def create(request):
-    if is_admin(request):
-        event_list = get_events_ordered_by_name()
-
-        if request.method == 'POST':
-            form = JobForm(request.POST)
-
-            if form.is_valid():
```
Comment (smarshy) [inline vms/job/views.py]: @amruthasangeeth are you checking for this validity clause in the new code?

Comment (amruthasangeeth) [inline vms/job/views.py]: @smarshy Yes, that is what we are doing using AdministratorLoginRequiredMixin

**Comment by smarshy** [comment]
@amruthasangeeth There are lots of errors in the build. It is a bit difficult to spot here where things might be going wrong.
Can you please try testing the changes once locally before that? Because then you would be able to spot easily if it is directing to an incorrect template or similar error. You can also run only the specific cases which are going wrong. It will fire up a browser and then you can observe what are the actions leading to the error. But since this might be a bit quick to observe, I would suggest trying to do it locally.

**Comment by amruthasangeeth** [comment]
@tapasweni-pathak All the test cases are passed

File: vms/job/templates/job/create.html [hunk 2]
```diff
@@ -49,7 +49,7 @@
 							<p class="form-control-static" id="end_date_here">end_date</p>
 						</div>
 					</div>
-                    <script src="{% static "vms/js/update-event-dates.js" %}"></script>
+                    <script src="{% static 'vms/js/update-event-dates.js' %}"></script>
                     {% if form.name.errors %}
```
Comment (tapasweni-pathak) [inline vms/job/templates/job/create.html]: @amruthasangeeth Why have we changed from double quote to single quote? Was this useful for you?

File: vms/job/views.py [hunk 3]
```diff
@@ -77,98 +99,133 @@ def create(request):
...
+        obj = Job.objects.get(pk=job_id)
+        return obj
+
+    def get_context_data(self, **kwargs):
+        context = super(JobUpdateView, self).get_context_data(**kwargs)
+        context['event_list'] = get_events_ordered_by_name()
+        return context
+
+    def form_valid(self, form):
+        job_id = self.kwargs['job_id']
```
Comment (tapasweni-pathak) [inline vms/job/views.py]: @amruthasangeeth Can't we reuse this method?

**Comment by amruthasangeeth** [comment]
@tapasweni-pathak @nida We need a code clean up here. There are unwanted lines of code in views.py like older FBV for creating job are still there.

**Comment by tapasweni-pathak** [comment]
@amruthasangeeth Can you please open a issue linking to the code that requires cleanup? I'll mark it for beginners. It will be useful for GCI.

**Comment by amruthasangeeth** [comment]
@tapasweni-pathak I have raised an issue for code clean up. Please mark it for beginners.

**Comment by tapasweni-pathak** [comment]
@amruthasangeeth Can you link the line from where it starts, in the issue?

**Comment by amruthasangeeth** [comment]
@tapasweni-pathak I have linked the unwanted lines. Can you mark it as first-timers?

**Pull request merged**